app/src/main/java/com/cronus/zdone/settings/SettingsView.kt

```diff
@@ -10,31 +10,30 @@ import kotlinx.android.synthetic.main.settings.view.*
 
 class SettingsView(context: Context) : BaseScreenView<SettingsScreen>(context) {
 
-    init {
-        inflate(context, R.layout.settings, this)
-        apiKeyEditText.addTextChangedListener(object : TextWatcher {
-            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
-            }
+    private val workTimeWatcher = object : TextWatcher {
```

Comment:
needed to extract these watchers so I could remove and re-add them after manually updating the text input fields. If we don't remove/re-add after manually updating, then these listeners fire and cause infinite-loop problems and general confusion.